doc/source/scripts/base.py

```diff
@@ -99,9 +96,7 @@ def sphinx_meth(meth, tilde=True):
         return ":meth:`{}{}.{}`".format(prefix, meth.__module__, meth.__qualname__)
 
     @staticmethod
-    def sphinx_ref(txt, label=None, suffix=""):
-        if label is None:
-            label = txt
+    def sphinx_ref(txt: str, label: str = None, suffix: str = "") -> str:
         return ":ref:`{} <{}{}>`".format(txt, label, suffix)
```

Comment:
As Rocco said in another PR, it'd be great to move solely onto f-strings; I think the user guide was started when we still supported 2.7, so there's a lot of now unnecessary contortions in the code. Out of scope here though!

Reply:
Let's do it here, it'd be simple enough